tests/test_check_runner.py

```diff
@@ -0,0 +1,145 @@
+"""Test suite for check_runner"""
+
+# pylint: disable=E1101, W0621, R0903
+
+import time
+from unittest.mock import call
+
+import pytest
+import cachetclient.cachet
+
+import check_runner as unit
+from monitoring_scripts.nagios_common import Codes
+from . import common
+
+
+class CompletedProcessMock():
+    """Mocks subprocess.CompletedProcess to test returncode"""
+
+    def __init__(self, returncode):
+        self.returncode = returncode
+
+
+@pytest.fixture(scope='module')
+def config(tmpdir_factory, base_url='http://some.url', api_key='namaste'):
+    """Fixture that generates a customisable config file"""
+
+    if base_url:
+        base_url_line = 'base-url: ' + base_url + '\n'
+    else:
+        base_url_line = ''
+    if api_key:
+        api_key_line = 'api-key: ' + api_key + '\n'
+    else:
+        api_key_line = ''
+
+    config_file = tmpdir_factory.mktemp('data').join('config.ini')
+    config_file.write('[Cachet]\n' + base_url_line + api_key_line)
+
+    return config_file.strpath
+
+
+@pytest.mark.parametrize(
+    'base_url,api_key',
+    [
+        (None, None),
+        (None, 'baby'),
+        ('donthurtme', None),
+    ]
+)
+def test_config_params(tmpdir_factory, mocker, base_url, api_key):
+    """Assert error is raised if a required parameter is missing from the config file"""
+
+    print('base_url = %s, api_key = %s', base_url, api_key)
+
+    config_file = config(tmpdir_factory, base_url=base_url, api_key=api_key)
+
+    with pytest.raises(ValueError):
```

Comment:
It could prove useful to assert keywords in the exception as well: https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

It's either that or create a dedicated exception for that error (which would inherit from `ValueError`). We wouldn't want that test to pass because it raises a `ValueError` somewhere else.